dash/include/dash/atomic/GlobAtomicRef.h

```diff
@@ -320,38 +321,80 @@ class GlobRef<dash::Atomic<T>>
    */
   T fetch_sub (
     /// Value to be subtracted from global atomic variable.
-    const T & value) const
+    const T & value)
   {
     return fetch_op(dash::plus<T>(), -value);
   }
 
-  /// prefix atomically increment value by one
-  T operator++ () const {
+  /**
+   * DASH specific variant which is faster than \c fetch_multiply
+   * but does not return value
+   */
+  void multiply(const T & value)
+  {
+    op(dash::multiply<T>(), value);
+  }
+
+  /**
+   * Atomic fetch-and-multiply operation on the referenced shared value.
+   *
+   * \return  The value of the referenced shared variable before the
+   *          operation.
+   */
+  T fetch_multiply(
+    /// Value to be added to global atomic variable.
+    const T & value)
+  {
+    return fetch_op(dash::multiply<T>(), value);
+  }
+
+  /**
+   * prefix atomically increment value by one
+   * Note that this operator does not return a reference but a copy of the value
+   * in order to ensure atomicity. This is consistent with the C++ std::atomic
+   * \c operator++, see
+   * http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith.
+   */
+  T operator++ () {
     return fetch_add(1) + 1;
   }
 
-  /// postfix atomically increment value by one
-  T operator++ (int) const {
+  /**
+   * postfix atomically increment value by one
+   */
+  T operator++ (int) {
     return fetch_add(1);
   }
 
-  /// prefix atomically decrement value by one
-  T operator-- () const {
-    return fetch_sub(1) - 1;
+  /**
+   * prefix atomically decrement value by one
+   * Note that this operator does not return a reference but a copy of the value
+   * in order to ensure atomicity. This is consistent with the C++ std::atomic
+   * \c operator--, see
+   * http://en.cppreference.com/w/cpp/atomic/atomic/operator_arith.
+   */
+  T operator-- () {
+    return fetch_sub(1) + 1;
   }
 
-  /// postfix atomically decrement value by one
-  T operator-- (int) const {
+  /**
+   * postfix atomically decrement value by one
+   */
+  T operator-- (int) {
     return fetch_sub(1);
   }
 
-  /// atomically increment value by ref
-  T operator+=(const T & value) const {
+  /**
+   * atomically increment value by ref
+   */
+  T operator+=(const T & value) {
     return fetch_add(value) + value;
   }
 
-  /// atomically decrement value by ref
-  T operator-=(const T & value) const {
+  /**
+   * atomically decrement value by ref
+   */
+  self_t& operator-=(const T & value) {
```

Comment:
Does this compile? Probably a missing unit test. Should be `T operator`

Reply:
Good catch, should be fixed now.